gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -790,14 +788,17 @@ _message_handler (void *thread_data)
     }
 
     if (result || gst_caps_can_intersect (client_caps, server_caps)) {
+      free (cmd_data.data.data);
       cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
       cmd_data.data.data = (uint8_t *) _sdata->sink_caps_str;
       cmd_data.data.size = (size_t) strlen (_sdata->sink_caps_str) + 1;
     } else {
       /* respond deny with src caps string */
       nns_loge ("Query caps is not acceptable!");
-      nns_loge ("Query client sink caps: %s", (char *) cmd_data.data.data);
+      nns_loge ("Query client sink caps: %s", cmd_data.data.data);
       nns_loge ("Query server src caps: %s", _sdata->src_caps_str);
+
+      free (cmd_data.data.data);
```

Comment:
move free() from LINE 791&801  to LINE 805.

Reply:
Is it okay not free cmd_data.data.data before LINE 793&803?
I thought it may occurs memory leak.
Could I know what I misunderstood?

Reply:
Basically, I agreed with MJ's opinion.

In this code, `cmd_data` is used for two purposes: Send & Receive Commands. And that makes the code more complicated to understand. What about using two different `cmd_data` for each purpose?

Reply:
I meant

Your code:
```
if (...) {
  ...
  free()
} else {
  ...
  free()
}
```

suggested code:
```
if (...) {
  ...
} else {
  ...
}
free()
```

Reply:
Thank you for a detailed description!